ligo/src/types.mligo

```diff
@@ -0,0 +1,235 @@
+// SPDX-FileCopyrightText: 2020 TQ Tezos
+// SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+// Corresponds to Types.hs module
+
+// -- FA2 types -- //
+
+#if !TYPES_H
+#define TYPES_H
+
+type token_id = nat
+type token_metadata =
+  [@layout:comb]
+  { token_id : token_id
+  ; symbol : string
+  ; name : string
+  ; decimals : nat
+  ; extras : (string, string) map
+  }
+
+type operator =
+  { owner : address
+  ; operator : address
+  }
```

Comment:
I'm not exactly sure why. I was testing the fa2 ligo contract and it seems like it needs `layout:comb` here.

```
type operator =
  [@layout:comb]
  { owner : address
  ; operator : address
  }
```

So do we really need this or maybe I did something wrong.

Reply:
Yes that look correct. The test for the matching types passes in both cases because they are both `address`, but without `layout:comb` LIGO reorders them alphabetically, so they end up inverted (compared to how we expect them to be).
Fixed now